streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -476,7 +485,23 @@ public StreamsConfig(final Map<?, ?> props) {
 
         return consumerProps;
     }
-
+                 
+    private void checkIfUnexpectedUserSpecifiedConsumerConfig(Map clientProvidedProps) {
+        // Streams do not allow users to configure certain consumer configurations, for example,
+        // enable_auto_commit. In cases where user tries to override such non-configurable
+        // consumer configurations, log a warning and remove the user defined value from the Map.
+        // Thus the default values for these consumer configurations that are suitable for
+        // streams will be used instead.
+        for (int index = 0; index < NON_CONFIGURABLE_CONSUMER_CONFIGS.length; index++) {
+            String consumerConfig = NON_CONFIGURABLE_CONSUMER_CONFIGS[index];
+            if (clientProvidedProps.containsKey(consumerConfig)) {
+                log.warn("Unexpected user-specified consumer config  " + consumerConfig
+                    + ", as the streams client will always use the default.");
+                clientProvidedProps.remove(consumerConfig);
```

Comment:
@mjsax Thanks for your comments. Appreciate it. I just had a quick question about your comment here. So what I am doing here is removing the non-configurable property from the clientProvidedProps map. The streams config defaults for this parameter is set back in the getCommonConsumerConfigs method at: https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java#L498 which is why I didn't set it again here. Is there something I am missing? Thanks @mjsax.